From 258663ace80f698af391bee605becdecca0e78c1 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 18 Jun 2026 17:30:26 +0200
Subject: [PATCH 13/15] lx2160acex7: clearfog-cx: activate linux netdevs during
 boot

Activate netdevs for sfp/qsfp by default during boot.

This is to mirror configuration of obp board as close as possible.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 .../clearfog-cx-s1_20-s2_0-dpl.dts            | 550 +++++++++-
 .../clearfog-cx-s1_8-s2_0-dpl.dts             | 986 ++++++++++++++++--
 2 files changed, 1471 insertions(+), 65 deletions(-)

diff --git a/config/lx2160a/LX2160A-CEX7/clearfog-cx-s1_20-s2_0-dpl.dts b/config/lx2160a/LX2160A-CEX7/clearfog-cx-s1_20-s2_0-dpl.dts
index 7831412..bf2f1c7 100644
--- a/config/lx2160a/LX2160A-CEX7/clearfog-cx-s1_20-s2_0-dpl.dts
+++ b/config/lx2160a/LX2160A-CEX7/clearfog-cx-s1_20-s2_0-dpl.dts
@@ -43,13 +43,13 @@
 				/* -------------- DPBPs --------------*/
 				obj_set@dpbp {
 					type = "dpbp";
-					ids = <0 1 2 3 4 5 >;
+					ids = <0 1 2 3 4 5 6 7 8 9 10 >;
 				};
 
 				/* -------------- DPCONs --------------*/
 				obj_set@dpcon {
 					type = "dpcon";
-					ids = <0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 >;
+					ids = <0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 >;
 				};
 
 				/* -------------- DPIOs --------------*/
@@ -67,13 +67,13 @@
 				/* -------------- DPMCPs --------------*/
 				obj_set@dpmcp {
 					type = "dpmcp";
-					ids = <1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 >;
+					ids = <1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 >;
 				};
 
 				/* -------------- DPNIs --------------*/
 				obj_set@dpni {
 					type = "dpni";
-					ids = <0 >;
+					ids = <0 1 2 3 4 5 >;
 				};
 
 				/* -------------- DPRTCs --------------*/
@@ -120,6 +120,26 @@
 			compatible = "fsl,dpbp";
 		};
 
+		dpbp@6 {
+			compatible = "fsl,dpbp";
+		};
+
+		dpbp@7 {
+			compatible = "fsl,dpbp";
+		};
+
+		dpbp@8 {
+			compatible = "fsl,dpbp";
+		};
+
+		dpbp@9 {
+			compatible = "fsl,dpbp";
+		};
+
+		dpbp@10 {
+			compatible = "fsl,dpbp";
+		};
+
 		dpcon@0 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
@@ -600,6 +620,406 @@
 			num_priorities = <0x2>;
 		};
 
+		dpcon@96 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@97 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@98 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@99 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@100 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@101 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@102 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@103 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@104 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@105 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@106 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@107 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@108 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@109 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@110 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@111 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@112 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@113 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@114 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@115 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@116 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@117 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@118 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@119 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@120 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@121 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@122 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@123 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@124 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@125 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@126 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@127 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@128 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@129 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@130 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@131 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@132 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@133 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@134 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@135 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@136 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@137 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@138 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@139 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@140 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@141 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@142 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@143 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@144 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@145 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@146 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@147 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@148 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@149 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@150 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@151 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@152 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@153 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@154 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@155 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@156 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@157 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@158 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@159 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@160 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@161 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@162 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@163 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@164 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@165 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@166 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@167 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@168 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@169 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@170 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@171 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@172 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@173 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@174 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@175 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
 		dpio@0 {
 			compatible = "fsl,dpio";
 			channel_mode = "DPIO_LOCAL_CHANNEL";
@@ -948,6 +1368,26 @@
 			compatible = "fsl,dpmcp";
 		};
 
+		dpmcp@58 {
+			compatible = "fsl,dpmcp";
+		};
+
+		dpmcp@59 {
+			compatible = "fsl,dpmcp";
+		};
+
+		dpmcp@60 {
+			compatible = "fsl,dpmcp";
+		};
+
+		dpmcp@61 {
+			compatible = "fsl,dpmcp";
+		};
+
+		dpmcp@62 {
+			compatible = "fsl,dpmcp";
+		};
+
 		dpni@0 {
 			compatible = "fsl,dpni";
 			type = "DPNI_TYPE_NIC";
@@ -963,6 +1403,81 @@
 			num_opr = <0>;
 		};
 
+		dpni@1 {
+			compatible = "fsl,dpni";
+			type = "DPNI_TYPE_NIC";
+			num_queues = <16>;
+			num_tcs = <1>;
+			num_cgs = <1>;
+			mac_filter_entries = <16>;
+			vlan_filter_entries = <0>;
+			fs_entries = <64>;
+			qos_entries = <0>;
+			dist_key_size = <56>;
+			num_channels= <1>;
+			num_opr = <0>;
+		};
+
+		dpni@2 {
+			compatible = "fsl,dpni";
+			type = "DPNI_TYPE_NIC";
+			num_queues = <16>;
+			num_tcs = <1>;
+			num_cgs = <1>;
+			mac_filter_entries = <16>;
+			vlan_filter_entries = <0>;
+			fs_entries = <64>;
+			qos_entries = <0>;
+			dist_key_size = <56>;
+			num_channels= <1>;
+			num_opr = <0>;
+		};
+
+		dpni@3 {
+			compatible = "fsl,dpni";
+			type = "DPNI_TYPE_NIC";
+			num_queues = <16>;
+			num_tcs = <1>;
+			num_cgs = <1>;
+			mac_filter_entries = <16>;
+			vlan_filter_entries = <0>;
+			fs_entries = <64>;
+			qos_entries = <0>;
+			dist_key_size = <56>;
+			num_channels= <1>;
+			num_opr = <0>;
+		};
+
+		dpni@4 {
+			compatible = "fsl,dpni";
+			type = "DPNI_TYPE_NIC";
+			num_queues = <16>;
+			num_tcs = <1>;
+			num_cgs = <1>;
+			mac_filter_entries = <16>;
+			vlan_filter_entries = <0>;
+			fs_entries = <64>;
+			qos_entries = <0>;
+			dist_key_size = <56>;
+			num_channels= <1>;
+			num_opr = <0>;
+		};
+
+		dpni@5 {
+			compatible = "fsl,dpni";
+			type = "DPNI_TYPE_NIC";
+			num_queues = <16>;
+			num_tcs = <1>;
+			num_cgs = <1>;
+			mac_filter_entries = <16>;
+			vlan_filter_entries = <0>;
+			fs_entries = <64>;
+			qos_entries = <0>;
+			dist_key_size = <56>;
+			num_channels= <1>;
+			num_opr = <0>;
+		};
+
 		dprtc@0 {
 			compatible = "fsl,dprtc";
 		};
@@ -979,8 +1494,33 @@
 	connections {
 
 		connection@1{
-			endpoint1 = "dpni@0";
+			endpoint1 = "dpni@5";
 			endpoint2 = "dpmac@17";
 		};
+
+		connection@2{
+			endpoint1 = "dpni@4";
+			endpoint2 = "dpmac@1";
+		};
+
+		connection@3{
+			endpoint1 = "dpni@3";
+			endpoint2 = "dpmac@7";
+		};
+
+		connection@4{
+			endpoint1 = "dpni@2";
+			endpoint2 = "dpmac@8";
+		};
+
+		connection@5{
+			endpoint1 = "dpni@1";
+			endpoint2 = "dpmac@9";
+		};
+
+		connection@6{
+			endpoint1 = "dpni@0";
+			endpoint2 = "dpmac@10";
+		};
 	};
 };
diff --git a/config/lx2160a/LX2160A-CEX7/clearfog-cx-s1_8-s2_0-dpl.dts b/config/lx2160a/LX2160A-CEX7/clearfog-cx-s1_8-s2_0-dpl.dts
index e91f837..0afe445 100644
--- a/config/lx2160a/LX2160A-CEX7/clearfog-cx-s1_8-s2_0-dpl.dts
+++ b/config/lx2160a/LX2160A-CEX7/clearfog-cx-s1_8-s2_0-dpl.dts
@@ -43,139 +43,811 @@
 				/* -------------- DPBPs --------------*/
 				obj_set@dpbp {
 					type = "dpbp";
+					ids = <0 1 2 3 4 5 6 7 8 >;
+				};
+
+				/* -------------- DPCONs --------------*/
+				obj_set@dpcon {
+					type = "dpcon";
+					ids = <0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 >;
+				};
+
+				/* -------------- DPIOs --------------*/
+				obj_set@dpio {
+					type = "dpio";
+					ids = <0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 >;
+				};
+
+				/* -------------- DPMACs --------------*/
+				obj_set@dpmac {
+					type = "dpmac";
+					ids = <3 4 5 6 7 8 9 10 17 >;
+				};
+
+				/* -------------- DPMCPs --------------*/
+				obj_set@dpmcp {
+					type = "dpmcp";
+					ids = <1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 >;
+				};
+
+				/* -------------- DPNIs --------------*/
+				obj_set@dpni {
+					type = "dpni";
+					ids = <0 1 2 3 4 5 6 7 8 >;
+				};
+
+				/* -------------- DPRTCs --------------*/
+				obj_set@dprtc {
+					type = "dprtc";
 					ids = <0 >;
 				};
 
-				/* -------------- DPCONs --------------*/
-				obj_set@dpcon {
-					type = "dpcon";
-					ids = <0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 >;
-				};
+				/* -------------- DPSECIs --------------*/
+				obj_set@dpseci {
+					type = "dpseci";
+					ids = <0 >;
+				};
+			};
+		};
+	};
+
+	/*****************************************************************
+	 * Objects
+	 *****************************************************************/
+	objects {
+
+		dpbp@0 {
+			compatible = "fsl,dpbp";
+		};
+
+		dpbp@1 {
+			compatible = "fsl,dpbp";
+		};
+
+		dpbp@2 {
+			compatible = "fsl,dpbp";
+		};
+
+		dpbp@3 {
+			compatible = "fsl,dpbp";
+		};
+
+		dpbp@4 {
+			compatible = "fsl,dpbp";
+		};
+
+		dpbp@5 {
+			compatible = "fsl,dpbp";
+		};
+
+		dpbp@6 {
+			compatible = "fsl,dpbp";
+		};
+
+		dpbp@7 {
+			compatible = "fsl,dpbp";
+		};
+
+		dpbp@8 {
+			compatible = "fsl,dpbp";
+		};
+
+		dpcon@0 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@1 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@2 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@3 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@4 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@5 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@6 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@7 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@8 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@9 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@10 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@11 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@12 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@13 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@14 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@15 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@16 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@17 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@18 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@19 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@20 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@21 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@22 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@23 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@24 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@25 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@26 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@27 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@28 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@29 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@30 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@31 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@32 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@33 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@34 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@35 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@36 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@37 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@38 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@39 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@40 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@41 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@42 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@43 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@44 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@45 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@46 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@47 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@48 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@49 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@50 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@51 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@52 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@53 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@54 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@55 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@56 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@57 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@58 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@59 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@60 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@61 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@62 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@63 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@64 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@65 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@66 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@67 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@68 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@69 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@70 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@71 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@72 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@73 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@74 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@75 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@76 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@77 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@78 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@79 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@80 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@81 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@82 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@83 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@84 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@85 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@86 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@87 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@88 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@89 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@90 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@91 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@92 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@93 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@94 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@95 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@96 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@97 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@98 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@99 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@100 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@101 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@102 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@103 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@104 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@105 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@106 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@107 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@108 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@109 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@110 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@111 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
 
-				/* -------------- DPIOs --------------*/
-				obj_set@dpio {
-					type = "dpio";
-					ids = <0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 >;
-				};
+		dpcon@112 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
 
-				/* -------------- DPMACs --------------*/
-				obj_set@dpmac {
-					type = "dpmac";
-					ids = <3 4 5 6 7 8 9 10 17 >;
-				};
+		dpcon@113 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
 
-				/* -------------- DPMCPs --------------*/
-				obj_set@dpmcp {
-					type = "dpmcp";
-					ids = <1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 >;
-				};
+		dpcon@114 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
 
-				/* -------------- DPNIs --------------*/
-				obj_set@dpni {
-					type = "dpni";
-					ids = <0 >;
-				};
+		dpcon@115 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
 
-				/* -------------- DPRTCs --------------*/
-				obj_set@dprtc {
-					type = "dprtc";
-					ids = <0 >;
-				};
+		dpcon@116 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
 
-				/* -------------- DPSECIs --------------*/
-				obj_set@dpseci {
-					type = "dpseci";
-					ids = <0 >;
-				};
-			};
+		dpcon@117 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
 		};
-	};
 
-	/*****************************************************************
-	 * Objects
-	 *****************************************************************/
-	objects {
+		dpcon@118 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
 
-		dpbp@0 {
-			compatible = "fsl,dpbp";
+		dpcon@119 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
 		};
 
-		dpcon@0 {
+		dpcon@120 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
 
-		dpcon@1 {
+		dpcon@121 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
 
-		dpcon@2 {
+		dpcon@122 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
 
-		dpcon@3 {
+		dpcon@123 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
 
-		dpcon@4 {
+		dpcon@124 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
 
-		dpcon@5 {
+		dpcon@125 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
 
-		dpcon@6 {
+		dpcon@126 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
 
-		dpcon@7 {
+		dpcon@127 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
 
-		dpcon@8 {
+		dpcon@128 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
 
-		dpcon@9 {
+		dpcon@129 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
 
-		dpcon@10 {
+		dpcon@130 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
 
-		dpcon@11 {
+		dpcon@131 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
 
-		dpcon@12 {
+		dpcon@132 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
 
-		dpcon@13 {
+		dpcon@133 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
 
-		dpcon@14 {
+		dpcon@134 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
 
-		dpcon@15 {
+		dpcon@135 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@136 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@137 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@138 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@139 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@140 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@141 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@142 {
+			compatible = "fsl,dpcon";
+			num_priorities = <0x2>;
+		};
+
+		dpcon@143 {
 			compatible = "fsl,dpcon";
 			num_priorities = <0x2>;
 		};
@@ -520,6 +1192,38 @@
 			compatible = "fsl,dpmcp";
 		};
 
+		dpmcp@53 {
+			compatible = "fsl,dpmcp";
+		};
+
+		dpmcp@54 {
+			compatible = "fsl,dpmcp";
+		};
+
+		dpmcp@55 {
+			compatible = "fsl,dpmcp";
+		};
+
+		dpmcp@56 {
+			compatible = "fsl,dpmcp";
+		};
+
+		dpmcp@57 {
+			compatible = "fsl,dpmcp";
+		};
+
+		dpmcp@58 {
+			compatible = "fsl,dpmcp";
+		};
+
+		dpmcp@59 {
+			compatible = "fsl,dpmcp";
+		};
+
+		dpmcp@60 {
+			compatible = "fsl,dpmcp";
+		};
+
 		dpni@0 {
 			compatible = "fsl,dpni";
 			type = "DPNI_TYPE_NIC";
@@ -531,6 +1235,128 @@
 			fs_entries = <64>;
 			qos_entries = <0>;
 			dist_key_size = <56>;
+			num_channels= <1>;
+			num_opr = <0>;
+		};
+
+		dpni@1 {
+			compatible = "fsl,dpni";
+			type = "DPNI_TYPE_NIC";
+			num_queues = <16>;
+			num_tcs = <1>;
+			num_cgs = <1>;
+			mac_filter_entries = <16>;
+			vlan_filter_entries = <0>;
+			fs_entries = <64>;
+			qos_entries = <0>;
+			dist_key_size = <56>;
+			num_channels= <1>;
+			num_opr = <0>;
+		};
+
+		dpni@2 {
+			compatible = "fsl,dpni";
+			type = "DPNI_TYPE_NIC";
+			num_queues = <16>;
+			num_tcs = <1>;
+			num_cgs = <1>;
+			mac_filter_entries = <16>;
+			vlan_filter_entries = <0>;
+			fs_entries = <64>;
+			qos_entries = <0>;
+			dist_key_size = <56>;
+			num_channels= <1>;
+			num_opr = <0>;
+		};
+
+		dpni@3 {
+			compatible = "fsl,dpni";
+			type = "DPNI_TYPE_NIC";
+			num_queues = <16>;
+			num_tcs = <1>;
+			num_cgs = <1>;
+			mac_filter_entries = <16>;
+			vlan_filter_entries = <0>;
+			fs_entries = <64>;
+			qos_entries = <0>;
+			dist_key_size = <56>;
+			num_channels= <1>;
+			num_opr = <0>;
+		};
+
+		dpni@4 {
+			compatible = "fsl,dpni";
+			type = "DPNI_TYPE_NIC";
+			num_queues = <16>;
+			num_tcs = <1>;
+			num_cgs = <1>;
+			mac_filter_entries = <16>;
+			vlan_filter_entries = <0>;
+			fs_entries = <64>;
+			qos_entries = <0>;
+			dist_key_size = <56>;
+			num_channels= <1>;
+			num_opr = <0>;
+		};
+
+		dpni@5 {
+			compatible = "fsl,dpni";
+			type = "DPNI_TYPE_NIC";
+			num_queues = <16>;
+			num_tcs = <1>;
+			num_cgs = <1>;
+			mac_filter_entries = <16>;
+			vlan_filter_entries = <0>;
+			fs_entries = <64>;
+			qos_entries = <0>;
+			dist_key_size = <56>;
+			num_channels= <1>;
+			num_opr = <0>;
+		};
+
+		dpni@6 {
+			compatible = "fsl,dpni";
+			type = "DPNI_TYPE_NIC";
+			num_queues = <16>;
+			num_tcs = <1>;
+			num_cgs = <1>;
+			mac_filter_entries = <16>;
+			vlan_filter_entries = <0>;
+			fs_entries = <64>;
+			qos_entries = <0>;
+			dist_key_size = <56>;
+			num_channels= <1>;
+			num_opr = <0>;
+		};
+
+		dpni@7 {
+			compatible = "fsl,dpni";
+			type = "DPNI_TYPE_NIC";
+			num_queues = <16>;
+			num_tcs = <1>;
+			num_cgs = <1>;
+			mac_filter_entries = <16>;
+			vlan_filter_entries = <0>;
+			fs_entries = <64>;
+			qos_entries = <0>;
+			dist_key_size = <56>;
+			num_channels= <1>;
+			num_opr = <0>;
+		};
+
+		dpni@8 {
+			compatible = "fsl,dpni";
+			type = "DPNI_TYPE_NIC";
+			num_queues = <16>;
+			num_tcs = <1>;
+			num_cgs = <1>;
+			mac_filter_entries = <16>;
+			vlan_filter_entries = <0>;
+			fs_entries = <64>;
+			qos_entries = <0>;
+			dist_key_size = <56>;
+			num_channels= <1>;
+			num_opr = <0>;
 		};
 
 		dprtc@0 {
@@ -549,8 +1375,48 @@
 	connections {
 
 		connection@1{
-			endpoint1 = "dpni@0";
+			endpoint1 = "dpni@8";
 			endpoint2 = "dpmac@17";
 		};
+
+		connection@2{
+			endpoint1 = "dpni@7";
+			endpoint2 = "dpmac@7";
+		};
+
+		connection@3{
+			endpoint1 = "dpni@6";
+			endpoint2 = "dpmac@8";
+		};
+
+		connection@4{
+			endpoint1 = "dpni@5";
+			endpoint2 = "dpmac@9";
+		};
+
+		connection@5{
+			endpoint1 = "dpni@4";
+			endpoint2 = "dpmac@10";
+		};
+
+		connection@6{
+			endpoint1 = "dpni@3";
+			endpoint2 = "dpmac@3";
+		};
+
+		connection@7{
+			endpoint1 = "dpni@2";
+			endpoint2 = "dpmac@4";
+		};
+
+		connection@8{
+			endpoint1 = "dpni@1";
+			endpoint2 = "dpmac@5";
+		};
+
+		connection@9{
+			endpoint1 = "dpni@0";
+			endpoint2 = "dpmac@6";
+		};
 	};
 };
-- 
2.51.0

